spvenvelope.go

```diff
@@ -0,0 +1,165 @@
+package bc
+
+import (
+	"context"
+	"errors"
+
+	"github.com/libsv/go-bt"
+)
+
+var (
+	// ErrPaymentNotVerified returns if a transaction in the tree provided was missed during verification
+	ErrPaymentNotVerified = errors.New("a tx was missed during validation")
+
+	// ErrTipTxConfirmed returns if the tip transaction is already confirmed
+	ErrTipTxConfirmed = errors.New("tip transaction must be unconfirmed")
```

Comment:
In some cases, it is possible or acceptable for an envelope to contain a single, already-confirmed transaction and no parents. One example use-case for a valid envelope that is already confirmed is when we want to store the merkle proof with a transaction that contains a UTXO which we are able to spend, so that it can later be furnished to the recipient of a future transaction.